app/models/blacklist.rb

```diff
@@ -0,0 +1,21 @@
+class Blacklist
+  def initialize(possible_match)
+    @possible_match = possible_match
+  end
+
+  def blacklisted?
+    if possible_match.nil?
+      false
+    else
+      host = URI.parse(possible_match).host
+
+      BlacklistedReferrer.where("host_with_path LIKE ?", "#{host}%").
+        or(BlacklistedReferrer.where("'www.'||host_with_path LIKE ?", "#{host}%")).
```

Comment:
Oh huh, what does the `||` do? Concatenate?

Reply:
Yep! I like it better than `SELECT CONCAT('www.', host_with_path)`.